pip/commands/list.py

```diff
@@ -122,10 +158,36 @@ def run(self, options, args):
             self.run_listing(options)
 
     def run_outdated(self, options):
+        latest_pkgs = []
         for dist, latest_version, typ in sorted(
                 self.find_packages_latest_versions(options),
                 key=lambda p: p[0].project_name.lower()):
             if latest_version > dist.parsed_version:
+                latest_pkgs.append((dist, latest_version, typ))
+
+        if (hasattr(options, "columns") and
```

Comment:
I'm comfortable with "get it working first, tidy up later". So don't worry about it for now.